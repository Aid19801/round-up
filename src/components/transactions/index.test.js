import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Transactions from './index';
import Title from '../../components/title';
import store from '../../store';

configure({ adapter: new Adapter() })

describe('Transactions', () => {
  
  it('should render the Transactions container without exploding...', () => {
      const wrapper = shallow(<Transactions />);
      expect(wrapper).toBeTruthy();
  });
  
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
            <Transactions />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it.skip('should render the expected children', () => {
    const el = <h4 className="loading">loading...</h4>;
    const wrapper = shallow(<Transactions isLoading />);
    expect(wrapper.find(el)).toHaveLength(1);
  });

  it.skip('should render the expected children', () => {
    const el = <h4>an error has occurred...</h4>;
    const wrapper = shallow(<Transactions error />);
    expect(wrapper.find(el)).toHaveLength(1);
  });
  
});