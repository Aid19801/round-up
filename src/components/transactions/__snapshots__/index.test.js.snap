// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions matches the snapshot 1`] = `
<div
  className="col-sm-6 transactions-container"
>
  <h4>
    Add To Savings: £
    0
  </h4>
  <button
    onClick={[Function]}
  >
    transfer =&gt; 
  </button>
  <div
    className="table-container"
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            £/GBP
          </th>
          <th
            scope="col"
          >
            Balance
          </th>
          <th
            scope="col"
          >
            Source
          </th>
          <th
            scope="col"
          >
            IN / OUT
          </th>
          <th
            scope="col"
          >
            Date
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
